### DISCUSSION TITLE
Can Elgato update the SDK used in the Objective-C and Swift examples (such as the Apple Mail)?

### DISCUSSION THREAD

**Discussion by aspdigital** [elgatosf/elgato#46] [Help]
I want to use my Streamdeck + to control a device connected to my Mac over RS-232. (Yes, really!) That hardware connects to the Mac with the usual USB-to-RS-232 converter dongle.  I have a macOS application that talks to that hardware, and I use the [ORSSerialPort](https://github.com/armadsen/ORSSerialPort) library to talk to the serial interface.

I can't use Javascript for this, as that language (I guess by design) cannot access hardware.  The plug-in system supports compiled languages, like C, C++, Objective-C and Swift, and the ORSSerialPort library supports both Objective-C and Swift. So it should be a simple matter of writing an Objective-C or Swift plug-in, right?

The [Changelog ](https://docs.elgato.com/sdk/plugins/changelog)indicates the Streamdeck SDK is up to version 6.1, but the examples that use non-Javascript languages use versions of the SDK that are lagging way behind. I noticed that the AppleMail plug-in example seems to conform to SDK version 2 (see the [defines](https://github.com/elgatosf/streamdeck-applemail/blob/master/Sources/Common/ESDSDKDefines.h)).  Confusingly, the commit history claims the SDK version it complies with is 4.3 but the definition in the header remains at 2. There is no definition for the StreamDeck +,  for starters.

Anyway.

That example was just updated an hour ago (!), but the only change made was a minor bug fix (and it was built as Universal and signed). The SDK part itself was not updated.  

Is there any hope that this Objective-C SDK will be updated to the current version 6.1?

**Comment by ChekTek** [comment]
Hi @aspdigital 

Unfortunately, there are no plans to update the Objective-C SDK. We only support the Typescript Node.js SDK at this time, however, there are a couple of options for native integrations. 

The recommended approach would be to create a native add-on using Node's built-in [N-API](https://nodejs.org/api/n-api.html). This would allow you to use C++ (or other low level languages) inside of the Node.js environment. We have plans for better documentation around how to do this, but unfortunately, they are not ready at this time.

There are also third-party SDKs that you could use to build native plugins although you may have to explore which ones are the most up to date. 

Lastly you could reference the underlying [WebSocket API](https://docs.elgato.com/streamdeck/sdk/references/websocket/plugin) that the new Typescript SDK uses to communication with Stream Deck.
